app/components/new-directory-user.js

```diff
@@ -132,10 +213,57 @@ export default Component.extend(NewUser, Validations, {
           isPresent(result.campusId);
         return result;
       });
-      this.set('tooManyResults', mappedResults.length > 50);
-      this.set('searchResults', mappedResults);
-      this.set('isSearching', false);
-      this.set('searchResultsReturned', true);
+      this.tooManyResults = mappedResults.length > 50;
+      this.searchResults = mappedResults;
+      this.searchResultsReturned = true;
+    }
+  }
+
+  @dropTask
+  *save() {
+    this.addErrorDisplaysFor([
+      'firstName',
+      'middleName',
+      'lastName',
+      'campusId',
+      'otherId',
+      'email',
+      'phone',
+      'username',
+      'password',
+    ]);
+    const isValid = yield this.isValid();
+    if (!isValid) {
+      return false;
+    }
+    const roles = yield this.store.findAll('user-role');
+    const primaryCohort = yield this.bestSelectedCohort;
```

Comment:
`this.bestSelectedCohort` is a getter and not async so doesn't need to be `yield`ed.

Reply:
:heavy_check_mark: